File: 程式語言 V2答案檔/Go/706.go
package main
import (
	"fmt"
	os "os"
	"bufio"
	"strconv"
	"sort"
)

func create_file(filename string) *os.File{
	f, err := os.Create("write.txt")
	if err != nil{
		panic(err)
	}
	return f
}

func output_text_to_file(f *os.File, text string){
	w := bufio.NewWriter(f)
	_, err := w.WriteString(text)
	if err != nil{
		panic(err)
	}
	w.Flush()
}

func main(){
	file, _ := os.Open("read.txt")
	sc := bufio.NewScanner(file)
	sc.Split(bufio.ScanLines)
	var number []int
	for i := 0; i < 4; i++{
		var val int
		fmt.Scanln(&val)
		number = append(number, val)
	}
	for sc.Scan() {
		val, _ := strconv.ParseInt(sc.Text(), 10, 64)
		number = append(number, int(val))
	}
	sort.Ints(number)
	output := ""
	for i := 0; i < len(number); i++ {
		output += strconv.Itoa(number[i]) + "\n"
	}
	f := create_file("write.txt")
	output_text_to_file(f, output)
}

File: 程式語言 V2答案檔/C++/706.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main()
{
    fstream input, output;
    int i, j, len = 0, number, * dataList;

    input.open("read.txt", ios::in);
    output.open("write.txt", ios::out);
    if (input.fail() || output.fail()) {
        cout << "檔案開啟失敗";
        return(0);
    }

    while (input >> number) {
        len++;
    }
    len = len + 4;
    dataList = new int[len];
    input.clear();
    input.seekg(0, input.beg);

    for (i = 0; i < 4; i++) {
        cin >> number;
        dataList[i] = number;
    }
    while (input >> number) {
        dataList[i] = number;
        i++;
    }

    for (i = 1; i < len; i++) {
        for (j = i; j > 0; j--) {
            if (dataList[j] < dataList[j - 1]) {
                number = dataList[j];
                dataList[j] = dataList[j - 1];
                dataList[j - 1] = number;
            }
        }
    }

    for (i = 0; i < len; i++) {
        cout << dataList[i] << endl;
        output << dataList[i] << endl;
    }

    input.close();
    output.close();
    delete[] dataList;
    return (0);
}